spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -218,32 +250,44 @@ private SpannerStatementQueryExecutor() {
 				keyParts.add(parentKeyParts.next());
 				tagNum++;
 			}
-			orJoiner.add(andJoiner.toString());
+			orParts.add(andJoiner.toString());
 		}
-		String cond = orJoiner.toString();
+		String keyClause = orParts.size() == 1
+				? orParts.get(0) : orParts.stream().map(s -> "(" + s + ")").collect(Collectors.joining(" OR "));
+		String condition = combine(keyClause, whereClause != null ? whereClause : "");
```

Comment:
let's move this ` whereClause != null ? whereClause : ""` to the `combine` method and apply it to both arguments